lib/ex_unit/lib/ex_unit.ex

```diff
@@ -360,6 +360,23 @@ defmodule ExUnit do
     ExUnit.Runner.run(options, nil)
   end
 
+  @doc """
+  Rerun the given test modules synchronously.
+
+  Returns a map containing the total number of tests, the number
+  of failures, the number of excluded tests and the number of skipped tests.
+  """
+  @spec rerun(list(atom())) :: suite_result()
+  def rerun(modules) do
+    for module <- modules do
+      ExUnit.Server.add_sync_module(module)
+    end
```

Comment:
Currently support rerunning it synchronously.

I tried to check the module attributes on `:ex_unit_async` on runtime to conditionally add the module sync/async. 

However, didn't have any luck with it as `Modules.get_attribute/3` won't work as the test module have been compiled. Tried using `module.__info__(:attributes)` as well, but upon inspecting I only have the `vsn` key:

```elixir
[vsn: [266934968422177934898710870860155740591]]
```

Probably need some pointers here 😅

Reply:
You will need to say that :ex_unit_async is a persisted attribute via `Module.register_attribute` :)

Reply:
Okay will do!